### PR TITLE
[BIT-626] Add neuron_info fields for ValidatorTrust and WeightConsensus

### PR DESCRIPTION
**Pull request by opentaco**
[BIT-626] Add neuron_info fields for ValidatorTrust and WeightConsensus

[BIT-626]: https://opentensor.atlassian.net/browse/BIT-626?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: bittensor/_subtensor/subtensor_mock.py [hunk 1]
```diff
@@ -49,6 +49,8 @@
                 ["rank", "u64"],
                 ["trust", "u64"],
                 ["consensus", "u64"],
+                ["validator_trust", "u64"],
+                ["weight_consensus", "u64"],
```
Comment (camfairchild) [inline bittensor/_subtensor/subtensor_mock.py]: aren't these u16?

File: bittensor/_wallet/wallet_impl.py [hunk 2]
```diff
@@ -87,6 +87,9 @@ def neuron(self, netuid: int) -> SimpleNamespace:
     def trust(self, netuid: int) -> SimpleNamespace:
         return self.get_neuron(netuid = netuid).trust
 
+    def validator_trust(self, netuid: int) -> SimpleNamespace:
+        return self.get_neuron(netuid = netuid).validator_trust
+
     def rank(self, netuid: int) -> SimpleNamespace:
         return self.get_neuron(netuid = netuid).rank
```
Comment (camfairchild) [inline bittensor/_wallet/wallet_impl.py]: The type hint for both should be `Optional[float]` or `float`. And possibly add a raises in a docstring if there's no None-neuron check

**Review by camfairchild — CHANGES_REQUESTED** [review]
Small changes, then LGTM

File: bittensor/_subtensor/chain_data.py [hunk 3]
```diff
@@ -113,11 +119,13 @@ def _neuron_dict_to_namespace(neuron_dict) -> 'NeuronInfo':
...
-            neuron.consensus = neuron.consensus / U64_MAX
-            neuron.incentive = neuron.incentive / U64_MAX
-            neuron.dividends = neuron.dividends / U64_MAX
+            neuron.rank = neuron.rank / U16_MAX
+            neuron.trust = neuron.trust / U16_MAX
+            neuron.consensus = neuron.consensus / U16_MAX
+            neuron.validator_trust = neuron.validator_trust / U16_MAX
+            neuron.weight_consensus = neuron.weight_consensus / U16_MAX
+            neuron.incentive = neuron.incentive / U16_MAX
+            neuron.dividends = neuron.dividends / U16_MAX
```
Comment (opentaco) [inline bittensor/_subtensor/chain_data.py:128]: @camfairchild all these needed to be corrected.

File: bittensor/_subtensor/subtensor_mock.py [hunk 4]
```diff
@@ -46,11 +46,13 @@
...
-                ["consensus", "u64"],
-                ["incentive", "u64"],
-                ["dividends", "u64"],
+                ["rank", "u16"],
+                ["trust", "u16"],
+                ["consensus", "u16"],
+                ["validator_trust", "u16"],
+                ["weight_consensus", "u16"],
+                ["incentive", "u16"],
+                ["dividends", "u16"],
```
Comment (opentaco) [inline bittensor/_subtensor/subtensor_mock.py:55]: @camfairchild all these needed correction as well.

File: bittensor/_wallet/wallet_impl.py [hunk 5]
```diff
@@ -81,48 +81,84 @@ def __str__(self):
...
+        neuron = self.get_neuron(netuid=netuid)
+        if neuron is None:
+            return None
+        return neuron.trust
+
+    def validator_trust(self, netuid: int) -> Optional[float]:
+        neuron = self.get_neuron(netuid=netuid)
+        if neuron is None:
+            return None
+        return neuron.validator_trust
```
Comment (opentaco) [inline bittensor/_wallet/wallet_impl.py:97]: @camfairchild most wallet_impl.py field methods needed type hint corrections and handling of possible None neuron.

File: bittensor/_wallet/wallet_impl.py [hunk 6]
```diff
@@ -81,48 +81,84 @@ def __str__(self):
...
-    def consensus(self, netuid: int) -> SimpleNamespace:
-        return self.get_neuron(netuid = netuid).consensus
+    def consensus(self, netuid: int) -> Optional[float]:
+        neuron = self.get_neuron(netuid=netuid)
+        if neuron is None:
+            return None
+        return neuron.consensus
 
-    def ip(self, netuid: int) -> SimpleNamespace:
-        return self.get_neuron(netuid = netuid).ip
```
Comment (opentaco) [inline bittensor/_wallet/wallet_impl.py:103]: @camfairchild removed wallet.ip since NeuronInfo doesn't contain it anymore.

**Review by camfairchild — APPROVED** [review]
LGTM. Thanks for the extra changes 🔥

**Pull request merged**